jsonlogic.go

```diff
@@ -519,6 +542,51 @@ func Apply(rule, data io.Reader, result io.Writer) error {
 	return json.NewEncoder(result).Encode(output)
 }
 
+func (e *Engine) Apply(rule, data io.Reader, result io.Writer) error {
+    if data == nil {
+        data = strings.NewReader("{}")
+    }
+
+    var _rule interface{}
+    var _data interface{}
+
+    ruleStr := new(strings.Builder)
+    _, err := io.Copy(ruleStr, rule)
+    if err != nil {
+        return err
+    }
+
+    ruleKey := ruleStr.String()
```

Comment:
Please hash the rules before use it as a cache key (SHA 256 would be good enough).

Why?

The rules could be huge and given enough rules being cached, it could end up using too much memory.

Reply:
Done @diegoholiveira